app.js

```diff
@@ -1,26 +1,66 @@
 const value = document.getElementById('passwordLength');
```

Comment:
Polecam już na początkowym etapie skonfigurować sobie TypeScripta w tym projekcie.
Żeby to zrobić effortless, wystarczy w konsoli będąc w root dir projektu wpisać `npm init`, co po zadaniu kilku pytań utworzy plik `package.json`. Dostaniesz pustą konfigurację. Żeby oszczędzić czas, możesz skorzystać z tego pliku:

`package.json`
```json
{
  "name": "PasswordGenerator",
  "version": "1.0.0",
  "description": "",
  "main": "app.js",
  "scripts": {
    "build": "tsc && cp -r public/. dist"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "@babel/cli": "^7.17.6",
    "@babel/core": "^7.17.7",
    "@babel/eslint-parser": "^7.17.0",
    "@babel/preset-env": "^7.16.11",
    "@babel/preset-typescript": "^7.16.7",
    "@types/node": "^17.0.21",
    "@typescript-eslint/eslint-plugin": "^5.15.0",
    "@typescript-eslint/parser": "^5.15.0",
    "babel-eslint": "^10.1.0",
    "eslint": "^8.11.0",
    "eslint-config-prettier": "^8.5.0",
    "eslint-plugin-import": "^2.25.4",
    "eslint-plugin-prettier": "^4.0.0",
    "prettier": "^2.6.0",
    "typescript": "^4.6.2"
  },
  "dependencies": {}
}
```

Jest tu już skonfiugurowany babel, eslint i prettier w wersjach współpracujących z TS, co zaoszczędzi Ci sporo zachodu.

Musisz też utworzyć pliki konfiguracyjne w głównym folderze projektu.

`tsconfig.json`
```json
{
  "compilerOptions": {
    "module": "commonjs",
    "esModuleInterop": true,
    "target": "es6",
    "moduleResolution": "node",
    "sourceMap": true,
    "outDir": "dist",
    "resolveJsonModule": true
  },
  "lib": ["es2015"]
}

```

`.eslintrc.js`
```js
module.exports = {
    root: true,
    env: {
        node: true,
    },
    extends: ['eslint:recommended', 'plugin:prettier/recommended'],
    parserOptions: {
        ecmaVersion: 2020,
        sourceType: 'module',
    },
    parser: '@typescript-eslint/parser',
    plugins: ['prettier', 'import', '@typescript-eslint'],
    rules: {
        'prettier/prettier': 'error',
        eqeqeq: ['error', 'smart'],
    },
};

```

`.prettierrc`
```json
{
    "tabWidth": 4,
    "printWidth": 120,
    "jsxBracketSameLine": true,
    "singleQuote": true,
    "trailingComma": "es5"
}

```

na sam koniec wpisz w terminalu `npm i` lub `npm install`. Środowisko będzie gotowe do pracy z typescriptem.

Możesz teraz zmienić nazwę pliku `app.js` na `app.ts`. 

Pliki `index.html` i `style.css` przenieś sobie do folderu `public`. 

Po odpaleniu w linii komend `npm run build` zadzieją się dwie rzeczy:
- `tsc` skompiluje twój plik `app.ts` i wszystkie pliki zależne do js i umieści je w folderze `dist`
- `cp -r public/. dist` skopiuje całą zawartość folderu `public` do folderu `dist`

Dzięki temu będziesz w folderze `dist` miał gotową do publikacji paczkę. 

Żeby nie zaśmiecać repo, warto folder `dist` dodać do `.gitignore`:

```
dist
node_modules

/idea
/.idea
```